js/components/interface/listViewer/ListViewer.js

```diff
@@ -0,0 +1,340 @@
+      onChange: conf.onChange === undefined ? undefined : this.preprocessAction(conf.onChange),
+      onKeyPress: conf.onKeyPress === undefined ? undefined : this.preprocessAction(conf.onKeyPress),
+      customComponent: conf.customComponent === undefined ? undefined : this.preprocessComponent(conf.customComponent),
+      configuration: conf.configuration === undefined ? undefined : this.preprocessColumnConfiguration(conf.configuration)
+    };
+
+  }
+
+  preprocessAction (action) {
+    if (isString(action)){
+      if (!this.handlerObject[action]){
+        throw new Error('Bad ListViewer configuration: the function ' + action + ' is not defined in the specified handler ' + this.handlerObject);
+      }
+      return entity => this.handlerObject[action](entity)
+    } else {
+      return action.bind(this.handlerObject);
+    } 
+  }
+
+  preprocessComponent (customComponent) {
+    if (isString(customComponent)) {
+      if (this.builtInComponents[customComponent]) {
+        return this.builtInComponents[customComponent];
+      } else if (window[customComponent]) {
+        return window[customComponent];
+      } else {
+        throw new Error('ListViewer configuration error: ' + customComponent + ' not defined. Try attach to the global (window) context or pass the imported object instead.');
+      }
+    }
+    return customComponent;
+  }
+
+  getFilterFn () {
+    return this.props.filterFn ? this.props.filterFn : () => true;
+  }
+
+  /**
+   * <ColumnDefinition key="path" id="path" customComponent={CustomColumn} />,
+   * <ColumnDefinition key="controls" id="actions" customHeadingComponent={CustomHeading} customComponent={CustomActions(buttonsConf)} />
+   * @param {*} param0 
+   */
+  getColumnDefinition (conf) {
+    let { id, customComponent, configuration, action } = conf;
+
+    if (configuration && customComponent) {
+      customComponent = customComponent(configuration);
+    }
+
+    if ( action && !customComponent) {
+      customComponent = WrapperComponent(action, customComponent);
+    }
+     
+    return React.createElement(ColumnDefinition, { ...conf, key: id, configuration: undefined, customComponent: customComponent, source: undefined });
+  }
+
+  getColumnDefinitions () {
+    return this.getColumnConfiguration().map(colConf => this.getColumnDefinition(colConf));
+  }
+
+  getLayout () {
+    return ({ Table, Pagination, Filter, SettingsWrapper }) => ( <div className="listviewer-container">
+      <Filter />
+      <Table />
+      <Pagination />
+    </div> );
+  }
+
+  render () {
+    window.conf = this.columnConfiguration;
+    return <section className="listviewer">
+ 
+      <Griddle
+
+        data={this.getData()}
+
+        plugins={this.props.infiniteScroll 
+          ? [plugins.LocalPlugin, plugins.PositionPlugin({})] 
+          : [plugins.LocalPlugin]
+        }
+
+        components={{ Layout: this.getLayout() }}
+        /*
```

Comment:
Of two minds if we should remove these commented out lines - it was your intention to keep them @filippomc ?